### ISSUE TITLE
OpCode0x19 doesn't set HALF_CARRY_FLAG as expected

### ISSUE THREAD

**Issue by dylmeadows** [darrena092/DMGe#38]
OpCode0x19 should affect the HALF_CARRY_FLAG but in the code it is only resetting the SUBTRACT_FLAG and the CARRY_FLAG. I'm not sure if this is an actual bug, but I did want to report it in case it is incorrect. Should be line 892-914.

**Comment by darrena092** [comment]
You are absolutely correct, it should be setting the half carry and carry flags according to the result of the operation. This is something I missed. I'll try to patch it when I get a chance, but I'm considering a complete rewrite of this emulator from the ground up.

It was done as a learning project, and lacks a lot of attention to detail.

**Comment by dylmeadows** [comment]
I'm working on writing one as well as a learning exercise, which is how I stumbled across your project and found the error when referencing how some of the opcodes work :) Would it be of any value for me to continue reporting any bugs I may find in the process of sifting through the opcodes?

**Comment by Quaker762** [comment]
Alright, I'm down for three people to work on this and actually get it working, plus @darrena092 I'm a lot more comfortable writing C/C++ now, so it'll be less spaghetti haha. I have a slightly updated version in plain C (in my repos), so if you guys want to pull anything from that, go ahead. Otherwise, let's just make a completely new repo and discuss stuff there.

I really need to filter DMGe emails so they get priority and I don't miss them...
